src/pike/smb2.py

```diff
@@ -4161,16 +4161,100 @@ def _decode(self, cur):
         self.security_mode = SecurityMode(cur.decode_uint16le())
         self.dialect = Dialect(cur.decode_uint16le())
 
+class AddressFamily(core.ValueEnum):
+    IPv4 = 0x0002
+    IPv6 = 0x0017
+
+AddressFamily.import_items(globals())
+
 
+class Capability(core.FlagEnum):
+     RSS_CAPABLE    = 0x00000001
+     RDMA_CAPABLE   = 0x00000002
+
+Capability.import_items(globals())
+
+class SockaddrIn:
+    def __init__(self):
+        port = array.array("B", [0] * 2)
+        ip = array.array("B", [0] * 4)
+        reserved = array.array("B", [0] * 8) 
+    def decode(self, cur):
+        self.port = cur.decode_uint16le()
+        self.ip = str(ipaddress.ip_address(cur.decode_uint32be())) 
+        self.reserved = cur.decode_uint64le()
+
+class SockaddrIn6:
+    def __init__(self):
+        port = array.array("B", [0] * 2)
+        flowinfo = array.array("B", [0] * 4)
+        ip = array.array("B", [0] * 16)
+        scopeid = array.array("B", [0] * 4)
+    def decode(self, cur):
+        self.port = cur.decode_uint16le()
+        self.flowinfo = cur.decode_uint32le()
+        self.ip = binascii.hexlify(cur.decode_bytes(16))
+        self.scopeid = cur.decode_uint32le()
+
+class SockaddrStorage(core.Frame):
+    def __init__(self, parent, sockaddr=None):
+        core.Frame.__init__(self, parent)
+        self.family = array.array("B", [0] * 2)
+        self.buffer = []
+        self.reserved = []
```

Comment:
the types for these initial values seem a bit weird.

i would expect them to be initialized to an empty SockaddrIn and empty byte array.